subprojects/core/src/main/java/org/gradle/internal/service/scopes/TaskExecutionServices.java

```diff
@@ -121,6 +127,7 @@ TaskExecuter createTaskExecuter(TaskArtifactStateRepository repository, TaskOutp
                                             packer,
                                             taskOutputsGenerationListener,
                                             taskOutputOriginFactory,
+                                            listenerManager,
```

Comment:
Can we handle the new listener the same way we handle `taskOutputsGenerationListener`? Passing the whole manager seems unnecessary here.